buildSrc/subprojects/buildquality/src/main/kotlin/org/gradle/gradlebuild/buildquality/VerifyBuildEnvironmentPlugin.kt

```diff
@@ -28,9 +28,10 @@ open class VerifyBuildEnvironmentPlugin : Plugin<Project> {
 
     private
     fun validateForAllCompileTasks(rootProject: Project) {
+        val availableJavaInstallations = rootProject.availableJavaInstallations
```

Comment:
This change means that these validation tasks would always validate the JVM that was used to serialise the task graph, not the JVM that is actually used. It would be better to model this using a service or build input, so that the details of the JVM are calculated on each build and validated. In the meantime, we should probably use the singleton+build finished listener pattern for this rather than using the root project to keep track of the state.